zebrad/tests/common/test_type.rs

```diff
@@ -152,18 +183,29 @@ impl TestType {
     /// Returns the Zebra state path for this test, if set.
     #[allow(clippy::print_stderr)]
     pub fn zebrad_state_path<S: AsRef<str>>(&self, test_name: S) -> Option<PathBuf> {
-        match env::var_os(ZEBRA_CACHE_DIR) {
-            Some(path) => Some(path.into()),
-            None => {
-                let test_name = test_name.as_ref();
+        // Load the effective config (defaults + optional TOML + env overrides)
+        let test_name = test_name.as_ref();
+        let cfg = match ZebradConfig::load(None) {
```

Comment:
We should either update the documentation for acceptance tests that required a cached state to say that they'll always run and look for a cached state in the default cache state directory path, or update this to check for the environment variable outside of `ZebradConfig::load()` so that it won't use the default value if no value is configured.